libsawtooth/src/client/rest.rs

```diff
@@ -117,6 +94,38 @@ impl SawtoothClient for RestApiSawtoothClient {
     }
 }
 
+/// used for deserializing single objects returned by the REST API.
+fn get<T>(url: &str) -> Result<Option<T>, SawtoothClientError>
+where
+    T: for<'a> serde::de::Deserialize<'a> + Sized,
+{
+    let request = Client::new().get(url);
+    let response = request
+        .send()
+        .map_err(|err| SawtoothClientError::new_with_source("request failed", err.into()))?;
+
+    if response.status().is_success() {
+        let obj: Single<T> = response.json().map_err(|err| {
+            SawtoothClientError::new_with_source("failed to deserialize response body", err.into())
+        })?;
+        Ok(Some(obj.data))
+    } else if response.status().as_u16() == 404 {
+        Ok(None)
+    } else {
+        let status = response.status();
+        let msg: ErrorResponse = response.json().map_err(|err| {
+            SawtoothClientError::new_with_source(
+                "failed to deserialize error response body",
+                err.into(),
+            )
+        })?;
+        Err(SawtoothClientError::new(&format!(
+            "failed to get item with given id: {}: {}",
```

Comment:
This message doesn't quite match the inputs to the format string.  This function doesn't know anything about IDs.  Perhaps a partial error message can be passed in, as a function.

Reply:
Created a function to get the type and id of the requested item and return a more accurate error message to be used in this function